src/main/java/scala_maven/HydraActivateLicenseMojo.java

```diff
@@ -0,0 +1,100 @@
+package scala_maven;
+
+import com.license4j.ActivationStatus;
+import com.license4j.License;
+import com.triplequote.license.HydraLicenseStore;
+import com.triplequote.license.LicenseMessages;
+import org.apache.maven.execution.MavenSession;
+import org.apache.maven.plugin.AbstractMojo;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.MojoFailureException;
+
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.nio.file.Paths;
+
+/**
+ * @goal hydraActivateLicense
+ * @requiresDirectInvocation true
+ */
+public class HydraActivateLicenseMojo extends AbstractMojo {
+
+    /**
+     * @parameter property="hydra.licenseKey"
+     */
+    private String licenseKey;
+
+    /**
+     * @parameter property="hydra.activationServerUrl" default-value="https://activation.triplequote.com/algas/"
+     *
+     */
+    private String activationServerUrl;
+
+    /**
+     * @parameter property="hydra.licenseStore" default-value="${user.home}/.triplequote/hydra.license"
+     *
+     */
+    private String licenseStore;
+
+    /**
+     * The Maven Session Object
+     *
+     * @parameter property="session"
+     * @required
+     * @readonly
+     */
+    protected MavenSession session;
+
+    @Override
+    public void execute() throws MojoExecutionException, MojoFailureException {
+        if (session.getCurrentProject() == session.getTopLevelProject()) {
+            try {
+                activateLicense();
+            } catch (MalformedURLException e) {
+                throw new MojoExecutionException(e, "Failed to parse URL", "");
+            }
```

Comment:
How about moving the `try...catch` where the URL exception can be thrown? (line 64 I presume)